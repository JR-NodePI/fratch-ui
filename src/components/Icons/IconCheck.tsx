import IconContainer from './components/IconContainer';
import { IconProps } from './components/IconProps';

const IconCheck = ({
  type,
  className,
}: Pick<IconProps, 'type' | 'className'>): JSX.Element => {
  return (
    <IconContainer className={className} type={type}>
      {({ iconClassName }) => (
        <svg
          className={iconClassName}
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="-1 -3 18 18"
        >
          <path
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2.2"
            d="M1 5.917 5.724 10.5 15 1.5"
          />
        </svg>
      )}
    </IconContainer>
  );
};

export default IconCheck;
