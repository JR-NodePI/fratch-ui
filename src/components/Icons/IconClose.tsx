import IconContainer from './components/IconContainer';
import { IconProps } from './components/IconProps';

const IconClose = ({
  type,
  className,
}: Pick<IconProps, 'type' | 'className'>): JSX.Element => {
  return (
    <IconContainer className={className} type={type}>
      {({ iconClassName }) => (
        <svg
          className={iconClassName}
          aria-hidden="true"
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="-1 -1 16 16"
        >
          <path
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
          />
        </svg>
      )}
    </IconContainer>
  );
};

export default IconClose;
