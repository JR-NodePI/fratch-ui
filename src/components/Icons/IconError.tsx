import IconContainer from './components/IconContainer';
import { IconProps } from './components/IconProps';

const IconError = ({
  type,
  className,
}: Pick<IconProps, 'type' | 'className'>): JSX.Element => {
  return (
    <IconContainer className={className} type={type}>
      {({ iconClassName }) => (
        <svg
          className={iconClassName}
          aria-hidden="true"
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 20 20"
        >
          <path
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="m13 7-6 6m0-6 6 6m6-3a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
          />
        </svg>
      )}
    </IconContainer>
  );
};

export default IconError;
