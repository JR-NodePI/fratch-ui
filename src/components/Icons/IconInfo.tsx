import IconContainer from './components/IconContainer';
import { IconProps } from './components/IconProps';

const IconInfo = ({
  type,
  className,
}: Pick<IconProps, 'type' | 'className'>): JSX.Element => {
  return (
    <IconContainer
      className={className}
      svg={
        <svg
          aria-hidden="true"
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 20 20"
        >
          <path
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="M8 9h2v5m-2 0h4M9.408 5.5h.01M19 10a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
          />
        </svg>
      }
      type={type}
    />
  );
};

export default IconInfo;
