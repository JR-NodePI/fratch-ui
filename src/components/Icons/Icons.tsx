export { default as IconArrowDown } from './Icons/IconArrowDown';
export { default as IconCheck } from './Icons/IconCheck';
export { default as IconClose } from './Icons/IconClose';
export { default as IconDownload } from './Icons/IconDownload';
export { default as IconEdit } from './Icons/IconEdit';
export { default as IconError } from './Icons/IconError';
export { default as IconExport } from './Icons/IconExport';
export { default as IconImport } from './Icons/IconImport';
export { default as IconInfo } from './Icons/IconInfo';
export { default as IconPause } from './Icons/IconPause';
export { default as IconPin } from './Icons/IconPin';
export { default as IconPlay } from './Icons/IconPlay';
export { default as IconPlus } from './Icons/IconPlus';
export { default as IconSuccess } from './Icons/IconSuccess';
export { default as IconUpload } from './Icons/IconUpload';
export { default as IconVerticalDots } from './Icons/IconVerticalDots';
export { default as IconWarning } from './Icons/IconWarning';
