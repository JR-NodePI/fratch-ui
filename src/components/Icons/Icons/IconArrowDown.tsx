import IconContainer from '../components/IconContainer';
import { IconProps } from './IconProps';

export default function IconArrowDown({
  type,
  className,
}: IconProps): JSX.Element {
  return (
    <IconContainer className={className} type={type}>
      {({ iconClassName }) => (
        <svg
          className={iconClassName}
          aria-hidden="true"
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="-1 -4 16 16"
        >
          <path
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="m1 1 5.326 5.7a.909.909 0 0 0 1.348 0L13 1"
          />
        </svg>
      )}
    </IconContainer>
  );
}
