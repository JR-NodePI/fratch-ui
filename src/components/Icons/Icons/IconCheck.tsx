import IconContainer from '../components/IconContainer';
import { IconProps } from './IconProps';

export default function IconCheck({ type, className }: IconProps): JSX.Element {
  return (
    <IconContainer className={className} type={type}>
      {({ iconClassName }) => (
        <svg
          className={iconClassName}
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="-1 -3 18 18"
        >
          <path
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2.2"
            d="M1 5.917 5.724 10.5 15 1.5"
          />
        </svg>
      )}
    </IconContainer>
  );
}
