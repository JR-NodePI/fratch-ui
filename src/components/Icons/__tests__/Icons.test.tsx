import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import * as icons from '../Icons';

describe('Icons', () => {
  describe('should render properly', () => {
    Object.entries(icons).forEach(([name, Icon]) => {
      it(name, () => {
        const { container } = render(<Icon />);
        expect(container).toMatchSnapshot();
      });
    });
  });
});
