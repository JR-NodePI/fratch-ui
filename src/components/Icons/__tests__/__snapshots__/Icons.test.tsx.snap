// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icons > should render properly > IconArrowDown 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="-1 -4 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m1 1 5.326 5.7a.909.909 0 0 0 1.348 0L13 1"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconCheck 1`] = `
<div>
  <svg
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="-1 -3 18 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 5.917 5.724 10.5 15 1.5"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="3"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconClose 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="-1 -1 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconDark 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 18 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.509 5.75c0-1.493.394-2.96 1.144-4.25h-.081a8.5 8.5 0 1 0 7.356 12.746A8.5 8.5 0 0 1 8.509 5.75Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconDownload 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 19"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 15h.01M4 12H2a1 1 0 0 0-1 1v4a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-4a1 1 0 0 0-1-1h-3M9.5 1v10.93m4-3.93-4 4-4-4"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconEdit 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 21 21"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.418 17.861 1 20l2.139-6.418m4.279 4.279 10.7-10.7a3.027 3.027 0 0 0-2.14-5.165c-.802 0-1.571.319-2.139.886l-10.7 10.7m4.279 4.279-4.279-4.279m2.139 2.14 7.844-7.844m-1.426-2.853 4.279 4.279"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconError 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m13 7-6 6m0-6 6 6m6-3a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconExport 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 18 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 6V2a.97.97 0 0 0-.933-1H5.828a2 2 0 0 0-1.414.586L1.586 4.414A2 2 0 0 0 1 5.828V18a.969.969 0 0 0 .933 1H14a1 1 0 0 0 1-1M6 1v4a1 1 0 0 1-1 1H1m6 6h9m-1.939-2.768L16.828 12l-2.767 2.768"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconImport 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 18 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 8V5.828a2 2 0 0 1 .586-1.414l2.828-2.828A2 2 0 0 1 7.828 1h8.239A.969.969 0 0 1 17 2v16a.969.969 0 0 1-.933 1H3.933A.97.97 0 0 1 3 18v-2M8 1v4a1 1 0 0 1-1 1H3m-2 6h10M9.061 9.232 11.828 12l-2.767 2.768"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconInfo 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 9h2v5m-2 0h4M9.408 5.5h.01M19 10a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconLight 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 3V1m0 18v-2M5.05 5.05 3.636 3.636m12.728 12.728L14.95 14.95M3 10H1m18 0h-2M5.05 14.95l-1.414 1.414M16.364 3.636 14.95 5.05M14 10a4 4 0 1 1-8 0 4 4 0 0 1 8 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconPause 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 12 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M1 1.8c0-.442.32-.8.714-.8h1.429c.394 0 .714.358.714.8v14.4c0 .442-.32.8-.714.8H1.714a.678.678 0 0 1-.505-.234A.851.851 0 0 1 1 16.2V1.8Zm7.143 0c0-.442.32-.8.714-.8h1.429c.19 0 .37.084.505.234.134.15.209.354.209.566v14.4c0 .442-.32.8-.714.8H8.857c-.394 0-.714-.358-.714-.8V1.8Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconPin 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="currentColor"
    viewBox="-20 -20 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      transform="rotate(22 200 200)"
    >
      <path
        d="M379.577,247.395c-14.944-18.75-31.833-28.12-50.679-28.12V73.092c9.896,0,18.47-3.616,25.693-10.848
       c7.238-7.233,10.858-15.802,10.858-25.697c0-9.9-3.62-18.464-10.858-25.697C347.368,3.617,338.795,0,328.898,0H146.179
       c-9.9,0-18.464,3.617-25.697,10.85c-7.234,7.233-10.85,15.796-10.85,25.697c0,9.899,3.616,18.464,10.85,25.697
       c7.233,7.232,15.8,10.848,25.697,10.848v146.179c-18.844,0-35.734,9.374-50.677,28.123c-14.941,18.75-22.414,39.824-22.414,63.242
       c0,4.948,1.809,9.232,5.426,12.847c3.619,3.613,7.901,5.421,12.85,5.421h115.344l21.698,138.474
       c0.95,5.141,3.996,7.707,9.135,7.707h0.291c2.274,0,4.235-0.808,5.851-2.423c1.615-1.619,2.614-3.572,2.994-5.855l14.558-137.902
       h122.482c4.948,0,9.236-1.808,12.847-5.421c3.621-3.614,5.432-7.898,5.432-12.847C401.999,287.222,394.519,266.145,379.577,247.395
       z M210.132,210.14c0,2.662-0.855,4.849-2.57,6.565c-1.713,1.714-3.903,2.57-6.567,2.57c-2.662,0-4.853-0.855-6.567-2.57
       c-1.713-1.713-2.568-3.903-2.568-6.565V82.233c0-2.664,0.855-4.858,2.568-6.567c1.714-1.711,3.905-2.568,6.567-2.568
       c2.664,0,4.854,0.854,6.567,2.568c1.714,1.712,2.57,3.903,2.57,6.567V210.14z"
      />
    </g>
  </svg>
</div>
`;

exports[`Icons > should render properly > IconPlay 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 16 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 1.984v14.032a1 1 0 0 0 1.506.845l12.006-7.016a.974.974 0 0 0 0-1.69L2.506 1.139A1 1 0 0 0 1 1.984Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconPlus 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="-0.5 -0.5 19 19"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 1v16M1 9h16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2.5"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconSuccess 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m7 10 2 2 4-4m6 2a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconUpload 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 19"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 15h.01M4 12H2a1 1 0 0 0-1 1v4a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-4a1 1 0 0 0-1-1h-3m-5.5 0V1.07M5.5 5l4-4 4 4"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconVerticalDots 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="currentColor"
    viewBox="0 0 4 15"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3.5 1.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Zm0 6.041a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Zm0 5.959a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Z"
    />
  </svg>
</div>
`;

exports[`Icons > should render properly > IconWarning 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 11V6m0 8h.01M19 10a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconArrowDown 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="-1 -4 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m1 1 5.326 5.7a.909.909 0 0 0 1.348 0L13 1"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconCheck 1`] = `
<div>
  <svg
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="-1 -3 18 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 5.917 5.724 10.5 15 1.5"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="3"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconClose 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="-1 -1 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconDark 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 18 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.509 5.75c0-1.493.394-2.96 1.144-4.25h-.081a8.5 8.5 0 1 0 7.356 12.746A8.5 8.5 0 0 1 8.509 5.75Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconDownload 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 19"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 15h.01M4 12H2a1 1 0 0 0-1 1v4a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-4a1 1 0 0 0-1-1h-3M9.5 1v10.93m4-3.93-4 4-4-4"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconEdit 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 21 21"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.418 17.861 1 20l2.139-6.418m4.279 4.279 10.7-10.7a3.027 3.027 0 0 0-2.14-5.165c-.802 0-1.571.319-2.139.886l-10.7 10.7m4.279 4.279-4.279-4.279m2.139 2.14 7.844-7.844m-1.426-2.853 4.279 4.279"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconError 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m13 7-6 6m0-6 6 6m6-3a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconExport 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 18 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 6V2a.97.97 0 0 0-.933-1H5.828a2 2 0 0 0-1.414.586L1.586 4.414A2 2 0 0 0 1 5.828V18a.969.969 0 0 0 .933 1H14a1 1 0 0 0 1-1M6 1v4a1 1 0 0 1-1 1H1m6 6h9m-1.939-2.768L16.828 12l-2.767 2.768"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconImport 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 18 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 8V5.828a2 2 0 0 1 .586-1.414l2.828-2.828A2 2 0 0 1 7.828 1h8.239A.969.969 0 0 1 17 2v16a.969.969 0 0 1-.933 1H3.933A.97.97 0 0 1 3 18v-2M8 1v4a1 1 0 0 1-1 1H3m-2 6h10M9.061 9.232 11.828 12l-2.767 2.768"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconInfo 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 9h2v5m-2 0h4M9.408 5.5h.01M19 10a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconLight 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 3V1m0 18v-2M5.05 5.05 3.636 3.636m12.728 12.728L14.95 14.95M3 10H1m18 0h-2M5.05 14.95l-1.414 1.414M16.364 3.636 14.95 5.05M14 10a4 4 0 1 1-8 0 4 4 0 0 1 8 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconPause 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 12 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M1 1.8c0-.442.32-.8.714-.8h1.429c.394 0 .714.358.714.8v14.4c0 .442-.32.8-.714.8H1.714a.678.678 0 0 1-.505-.234A.851.851 0 0 1 1 16.2V1.8Zm7.143 0c0-.442.32-.8.714-.8h1.429c.19 0 .37.084.505.234.134.15.209.354.209.566v14.4c0 .442-.32.8-.714.8H8.857c-.394 0-.714-.358-.714-.8V1.8Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconPin 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="currentColor"
    viewBox="-20 -20 500 500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      transform="rotate(22 200 200)"
    >
      <path
        d="M379.577,247.395c-14.944-18.75-31.833-28.12-50.679-28.12V73.092c9.896,0,18.47-3.616,25.693-10.848
       c7.238-7.233,10.858-15.802,10.858-25.697c0-9.9-3.62-18.464-10.858-25.697C347.368,3.617,338.795,0,328.898,0H146.179
       c-9.9,0-18.464,3.617-25.697,10.85c-7.234,7.233-10.85,15.796-10.85,25.697c0,9.899,3.616,18.464,10.85,25.697
       c7.233,7.232,15.8,10.848,25.697,10.848v146.179c-18.844,0-35.734,9.374-50.677,28.123c-14.941,18.75-22.414,39.824-22.414,63.242
       c0,4.948,1.809,9.232,5.426,12.847c3.619,3.613,7.901,5.421,12.85,5.421h115.344l21.698,138.474
       c0.95,5.141,3.996,7.707,9.135,7.707h0.291c2.274,0,4.235-0.808,5.851-2.423c1.615-1.619,2.614-3.572,2.994-5.855l14.558-137.902
       h122.482c4.948,0,9.236-1.808,12.847-5.421c3.621-3.614,5.432-7.898,5.432-12.847C401.999,287.222,394.519,266.145,379.577,247.395
       z M210.132,210.14c0,2.662-0.855,4.849-2.57,6.565c-1.713,1.714-3.903,2.57-6.567,2.57c-2.662,0-4.853-0.855-6.567-2.57
       c-1.713-1.713-2.568-3.903-2.568-6.565V82.233c0-2.664,0.855-4.858,2.568-6.567c1.714-1.711,3.905-2.568,6.567-2.568
       c2.664,0,4.854,0.854,6.567,2.568c1.714,1.712,2.57,3.903,2.57,6.567V210.14z"
      />
    </g>
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconPlay 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 16 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 1.984v14.032a1 1 0 0 0 1.506.845l12.006-7.016a.974.974 0 0 0 0-1.69L2.506 1.139A1 1 0 0 0 1 1.984Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconPlus 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="-0.5 -0.5 19 19"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 1v16M1 9h16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2.5"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconSuccess 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m7 10 2 2 4-4m6 2a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconUpload 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    viewBox="0 0 20 19"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 15h.01M4 12H2a1 1 0 0 0-1 1v4a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-4a1 1 0 0 0-1-1h-3m-5.5 0V1.07M5.5 5l4-4 4 4"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconVerticalDots 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="currentColor"
    viewBox="0 0 4 15"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3.5 1.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Zm0 6.041a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Zm0 5.959a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Z"
    />
  </svg>
</div>
`;

exports[`Icons.ts > should render properly > IconWarning 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="_icon_2be8fc _primary_2be8fc"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 11V6m0 8h.01M19 10a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;
